pkg/apiserver/debugapi/param_model_def_test.go

```diff
@@ -0,0 +1,167 @@
+}
+
+func (t *testParamModelsSuite) Test_APIParamModelMultiTags(c *C) {
+	client := endpoint.NewClient(&testFetcher{})
+	testEndpoint := &endpoint.APIModel{
+		ID:        "test_endpoint",
+		Component: model.NodeKindTiDB,
+		Path:      "/test",
+		Method:    http.MethodGet,
+		QueryParams: []*endpoint.APIParam{
+			{
+				Name:  "param1",
+				Model: APIParamModelTags,
+			},
+		},
+	}
+	client.RegisterEndpoint([]*endpoint.APIModel{testEndpoint})
+	value1 := url.QueryEscape("value1,,, ")
+	value2 := url.QueryEscape("value2")
+	param1 := fmt.Sprintf("%s,%s", value1, value2)
+
+	req, err := client.Send(testEndpoint.ID, "127.0.0.1", 10080, map[string]string{
+		"param1": param1,
+	})
+	if err != nil {
+		c.Error(err)
+	}
+	data, _ := req.Body()
+
+	c.Assert(string(data), Equals, testCombineReq("127.0.0.1", 10080, "/test", fmt.Sprintf("param1=%s&param1=%s", value1, value2)))
+}
+
+func (t *testParamModelsSuite) Test_APIParamModelInt(c *C) {
+	client := endpoint.NewClient(&testFetcher{})
+	testEndpoint := &endpoint.APIModel{
+		ID:        "test_endpoint",
+		Component: model.NodeKindTiDB,
+		Path:      "/test",
+		Method:    http.MethodGet,
+		QueryParams: []*endpoint.APIParam{
+			{
+				Name:  "param1",
+				Model: APIParamModelInt,
+			},
+		},
+	}
+	client.RegisterEndpoint([]*endpoint.APIModel{testEndpoint})
+
+	_, err := client.Send(testEndpoint.ID, "127.0.0.1", 10080, map[string]string{
+		"param1": "value1",
+	})
+	c.Log(err)
+	c.Assert(errorx.IsOfType(err, endpoint.ErrInvalidParam), Equals, true)
+
+	req2, err := client.Send(testEndpoint.ID, "127.0.0.1", 10080, map[string]string{
+		"param1": "2",
+	})
+	if err != nil {
+		c.Error(err)
+	}
+	data, _ := req2.Body()
+	c.Assert(string(data), Equals, testCombineReq("127.0.0.1", 10080, "/test", "param1=2"))
+}
+
+func (t *testParamModelsSuite) Test_APIParamModelConstant(c *C) {
+	client := endpoint.NewClient(&testFetcher{})
+	testEndpoint := &endpoint.APIModel{
+		ID:        "test_endpoint",
+		Component: model.NodeKindTiDB,
+		Path:      "/test",
+		Method:    http.MethodGet,
+		QueryParams: []*endpoint.APIParam{
+			{
+				Name:  "param1",
+				Model: APIParamModelConstant("value1"),
+			},
+		},
+	}
+	client.RegisterEndpoint([]*endpoint.APIModel{testEndpoint})
+
+	req, err := client.Send(testEndpoint.ID, "127.0.0.1", 10080, map[string]string{})
+	if err != nil {
+		c.Error(err)
+	}
+	data, _ := req.Body()
+	c.Assert(string(data), Equals, testCombineReq("127.0.0.1", 10080, "/test", "param1=value1"))
+}
+
+func (t *testParamModelsSuite) Test_APIParamModelEnum(c *C) {
+	client := endpoint.NewClient(&testFetcher{})
+	value1 := "value1"
+	testEndpoint := &endpoint.APIModel{
+		ID:        "test_endpoint",
+		Component: model.NodeKindTiDB,
+		Path:      "/test",
+		Method:    http.MethodGet,
+		QueryParams: []*endpoint.APIParam{
+			{
+				Name:  "param1",
+				Model: APIParamModelEnum([]EnumItem{{Value: value1}}),
+			},
+		},
+	}
+	client.RegisterEndpoint([]*endpoint.APIModel{testEndpoint})
```

Comment:
Seems that endpoint and API should be unified to one, reducing number of concepts.